app/models/grda_warehouse/data_source.rb

```diff
@@ -574,6 +574,33 @@ def hmis_url_for(entity)
     # For any other Enrollment-related record, link to the enrollment page
     url ||= "#{base}/client/#{entity.client&.id}/enrollments/#{entity.enrollment&.id}" if entity.respond_to?(:enrollment) && entity.respond_to?(:client)
 
+    # If we don't have the HMIS driver we probably aren't here, but we need to check for the next section
+    # If we don't have a user, just return the URl (backwards compatibility)
+    return url if user.blank?
+
+    # If we have a user, check for access (on the HMIS side)
+    hmis_entity = if entity.respond_to?(:as_hmis)
+      entity.as_hmis(force_reload: false)
+    else
+      entity
+    end
+    hmis_user = user.related_hmis_user(self)
+
+    known_permissions = {
+      'GrdaWarehouse::Hud::Project' => [:can_view_project],
+      # 'GrdaWarehouse::Hud::Organization' => [:can_view_organization],
+      'GrdaWarehouse::Hud::Client' => [:can_view_clients],
+      'Hmis::Hud::Enrollment' => [:can_view_projects, :can_view_enrollment_details],
```

Comment:
```suggestion
      'GrdaWarehouse::Hud::Enrollment' => [:can_view_projects, :can_view_enrollment_details],
```
I dont think there is any place we are passing an HMIS enrollment to this? but I could be wrong

Reply:
I'm specifically looking at https://hmis-warehouse.dev.test/source_data/59064?type=Enrollment which is a grdawarehouse model